File: mobile/lib/types/training.ts
export type TrainingPlanSourceType = "system" | "ai" | "coach";
export type TrainingPlanStatus = "active" | "completed" | "archived";
export type AssignmentStatus = "pending" | "in_progress" | "completed" | "skipped";

export interface TrainingPlan {
  id: string;
  student_id: string;
  sport_id: string;
  source_type: TrainingPlanSourceType;
  title: string;
  description: string | null;
  start_date: string | null;
  end_date: string | null;
  status: TrainingPlanStatus;
  created_at: string;
}

export interface TrainingAssignment {
  id: string;
  training_plan_id: string;
  student_id: string;
  day_number: number | null;
  scheduled_date: string | null;
  drill_id: string;
  sets: number | null;
  reps: number | null;
  status: AssignmentStatus;
  notes: string | null;
}

File: backend/app/schemas/training.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel

from app.models.training_assignment import AssignmentStatus
from app.models.training_plan import TrainingPlanSourceType, TrainingPlanStatus


class TrainingPlanRead(BaseModel):
    id: uuid.UUID
    student_id: uuid.UUID
    sport_id: uuid.UUID
    source_type: TrainingPlanSourceType
    title: str
    description: str | None = None
    start_date: date | None = None
    end_date: date | None = None
    status: TrainingPlanStatus
    created_at: datetime

    class Config:
        from_attributes = True


class TrainingAssignmentRead(BaseModel):
    id: uuid.UUID
    training_plan_id: uuid.UUID
    student_id: uuid.UUID
    day_number: int | None = None
    scheduled_date: date | None = None
    drill_id: uuid.UUID
    sets: int | None = None
    reps: int | None = None
    status: AssignmentStatus
    notes: str | None = None

    class Config:
        from_attributes = True


class TrainingAssignmentUpdate(BaseModel):
    status: AssignmentStatus | None = None
    notes: str | None = None
